pkg/chunkenc/pool.go

```diff
@@ -359,7 +359,7 @@ func (pool *SnappyPool) GetWriter(dst io.Writer) io.WriteCloser {
 		writer.Reset(dst)
 		return writer
 	}
-	return snappy.NewBufferedWriter(dst)
+	return snappy.NewWriter(dst, snappy.WriterSnappyCompat(), snappy.WriterBlockSize(4096))
```

Comment:
`s2.WriterBlockSize(4096)`, which is the minimum block size, get's us the best performance across all benchmarks but leads to the worst case scenario for `BenchmarkWrite/ordered-snappy-32`